internal/services/containers/container_registry_credential_set_resource_test.go

```diff
@@ -0,0 +1,173 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package containers_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/containerregistry/2023-07-01/credentialsets"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance/check"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+)
+
+type ContainerRegistryCredentialSetResource struct{}
+
+func TestAccContainerRegistryCredentialSet_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_container_registry_credential_set", "test")
+	r := ContainerRegistryCredentialSetResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccContainerRegistryCredentialSet_requiresImport(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_container_registry_credential_set", "test")
+	r := ContainerRegistryCredentialSetResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		{
+			Config:      r.requiresImport(data),
+			ExpectError: acceptance.RequiresImportError("azurerm_container_registry_credential_set"),
+		},
```

Comment:
We can replace this with 
```suggestion
		data.RequiresImportErrorStep(r.requiresImport),
```